### PR TITLE
Updates Async extern

### PR DESCRIPTION
**Pull request by JbIPS**

### CODE REVIEW

**Comment by fullofcaffeine** [comment]
What's `Async` for? It can be easily confused with the @async metadata

**Comment by JbIPS** [comment]
From [Async doc](https://caolan.github.io/async/docs.html):
> Async is a utility module which provides straight-forward, powerful functions for working with asynchronous JavaScript.

File: js/npm/Async.hx [hunk 1]
```diff
@@ -1,42 +1,65 @@
...
+import haxe.extern.EitherType;
+import haxe.extern.Rest;
 
 typedef AsyncCallback<T> = T -> Void;
 typedef AsyncOptionalCallback<T> = ?T -> Void;
 typedef AsyncMapCallback<Err, T> = Err -> T -> Void;
 
 typedef AsyncLoopCallback<Err> = AsyncCallback<Err> -> Void;
+typedef TaskFunction<T, Err> = ?T -> (Null<Err> -> T);
+typedef Task<T, Err> = EitherType<TaskFunction<T, Err>, Array<EitherType<String, TaskFunction<T, Err>>>>;
```
Comment (clemos) [inline js/npm/Async.hx]: please prefix, like `AsyncTask`, `AsyncTaskFunction`, etc.

File: js/npm/Async.hx [hunk 2]
```diff
@@ -116,3 +150,23 @@ implements npm.Package.Require<"async", "^0.9.0">
 
 	public static function noConflict() : Class<Async>;
 }
+
+extern class CargoObject<T, Err> {
```
Comment (clemos) [inline js/npm/Async.hx]: please prefix `AsyncCargoObject`

File: js/npm/Async.hx (hunk 1 above)
Comment (clemos) [inline js/npm/Async.hx]: also, maybe it would be nice to standardize `Err, T` order.

File: js/npm/Async.hx [hunk 3]
```diff
@@ -112,9 +112,9 @@ implements npm.Package.Require<"async", "^2.3.0">
 
 	// queue
 	// priorityQueue
-	public static function cargo<T, Err>(worker: AsyncMapCallback<T, Err>, payload: Int) : CargoObject<T, Err>;
+	public static function cargo<T, Err>(worker: AsyncMapCallback<T, Err>, payload: Int) : AsyncCargoObject<T, Err>;
```
Comment (clemos) [inline js/npm/Async.hx]: `Err, T`

**Pull request merged**